### ISSUE TITLE
Tasks when POA changes in BGS

### ISSUE THREAD

**Issue by laurjpeterson** [department-of-veterans-affairs/caseflow#7784]
Veterans can changes their VSO at any point in the appeals process. This will affect VSOs that are already assigned tasks. We propose a [nightly job that checks POA](https://github.com/department-of-veterans-affairs/caseflow/issues/8336). This issue catalogues requirements after that POA change is detected.  

### Requirements
- [x] The correct, new VSO is assigned the IHP task and it appears in the team queue.
- [x] The former VSO tasks are canceled. 
- [x] A new `TrackVeteran` task is created. 
- [ ] The new VSO has the same IHP time period as the original VSO. 
   - Because of this requirement, I don't think we can simply reassign the task. We need the IHP days waiting to start at 0 for the new VSO. 
- [ ] The previous VSO team and individual (if applicable) are notified that this case is no longer on their plate. (needed?)
- [ ] If the case had already been distributed to a judge (or later), the case needs to be given back to the VSO for the opportunity to write the IHP. 
    - Option 1: Caseflow could place the in progress task (i.e. JudgeAssignTask) on hold until the IHP task is complete, so the case doesn't need to be redistributed. 
    - Option 2: Caseflow could add the IHP task to the root and close/change all tasks so the case could be distributed again when the IHP time period is over (like a normal case).
    - Whatever implementation we go for should take Case Timeline into account, so the story of the appeal is clear

## Blocked by:
- [x] Creation of VSO `TrackVeteranTask` (#7905)
- [x] Add ability to cancel tasks (#8701)

**Comment by lowellrex** [comment]
Blocked by #7905.

**Comment by laurjpeterson** [comment]
Looks like both tickets that were blocking this one have been closed. @lowellrex - can we begin working on this (if prioritized for a sprint)? Do we have any solution for POA changes in BGS in place today?

**Comment by lowellrex** [comment]
When POA changes for a VSO [the following already happens](https://github.com/department-of-veterans-affairs/caseflow/blob/2d6fc81174a51c640e5e57b04027940fc46a26e2/app/models/tasks/track_veteran_task.rb#L33):
* We cancel any existing tasks assigned to the old VSO (tracking and IHP tasks)
* We create a `TrackVeteranTask` for the new VSO and an IHP task if the AMA appeal is in the hearing docket

According to my reading of the requirements for this ticket the following still needs to happen:
* Pause the deciding of the appeal if it has already been distributed to a judge (either by placing the existing tasks on hold while the IHP is written, or by cancelling the existing tasks and redistributing the appeal when the IHP has been written)
* (Maybe) notify the previous VSO that they were removed as the representative

Is that correct?

**Comment by laurjpeterson** [comment]
Thank you! Ok, will keep this open.

**Comment by lowellrex** [comment]
How do we want to pause the deciding of the appeal? By placing the existing tasks on hold while the IHP is written, or by cancelling the existing tasks and redistributing the appeal when the IHP has been written?

**Comment by laurjpeterson** [comment]
I can confirm with BVA, but I think they'll choose `By placing the existing tasks on hold while the IHP is written` because the judge/attorney team may have already started working on the appeal. Should we create a separate ticket for this?

**Comment by lowellrex** [comment]
> Should we create a separate ticket for this?

Done! #10800
